public/app/features/datasources/DataSourcesListPage.tsx

```diff
@@ -58,12 +70,29 @@ export class DataSourcesListPage extends PureComponent<Props> {
     return await this.props.loadDataSources();
   }
 
+  deleteDataSource = (id: number) => {
+    appEvents.emit(CoreEvents.showConfirmModal, {
+      title: 'Delete',
+      text: 'Are you sure you want to delete this data source?',
+      yesText: 'Delete',
+      icon: 'fa-trash',
+      onConfirm: () => {
+        this.confirmDeleteDataSource(id);
+      },
+    });
+  };
+
+  confirmDeleteDataSource = async (id: number) => {
+    await getBackendSrv()
+      .delete(`/api/datasources/${id}`)
+      .then(() => this.fetchDataSources());
+  };
```

Comment:
This should use the redux action deleteDatasource instead (but modify that action to take in an id), as that action aslo updates the frontend cache of available data sources.